### ISSUE TITLE
SonosNetwork.js bug

### ISSUE THREAD

**Issue by crmunro** [sonos-web/sonos-web#41]
Hi there! 

I'm running into a problem running Sonos-Web: upon start-up, the UI fails to load and throws the following error: 
```
TypeError: Cannot read property 'forEach' of undefined
    at SonosNetwork._getQueue (/Users/crmunro/.sonos-web/src/sonos/SonosNetwork.js:715:19)
    at processTicksAndRejections (internal/process/next_tick.js:81:5)
```
I can comment out the corresponding lines of code, and the web app loads fine but only allows interacting with the play/pause/volume features, we cannot browse for other music or view the queue. This is while we are playing music from a Spotify Connect session.

I'm wondering if this has something to do with a bugfix in the latest version of [node-sonos](https://github.com/bencevans/node-sonos/releases), which relates to some missing web pages on the Sonos impacting the ability to get a Spotify Connect queue. 

Any thoughts or things that I can try? Thanks!

**Comment by ghost** [comment]
I too have this issue (running Ubuntu 19.10)

**Comment by puzio0** [comment]
Same issue here:  version 1.3.1

**Comment by martinfowler** [comment]
I just got this too - again version 1.3.1 on node v12.13.1 on Ubuntu 18.04.4 LTS

I tried uninstalling and reinstalling sonos-web - didn't help

**Comment by Quizli** [comment]
Same here :-( Even slightly longer error msg:

> TypeError: Cannot read property 'forEach' of undefined\n at SonosNetwork._getQueue (C:\\Users\\admin\\.sonos-web\\src\\sonos\\SonosNetwork.js:715:19)\n at runMicrotasks (<anonymous>)\n at processTicksAndRejections (internal/process/task_queues.js:97:5)\n at async SonosNetwork._getAVTransportInfo (C:\\Users\\admin\\.sonos-web\\src\\sonos\\SonosNetwork.js:810:19)\n at async C:\\Users\\admin\\.sonos-web\\src\\sonos\\SonosNetwork.js:624:33\n at async Promise.all (index 1)\n at async C:\\Users\\admin\\.sonos-web\\src\\sonos\\SonosNetwork.js:623:9","message":"Cannot read property 'forEach' of undefined

Version 1.3.1 and Windows

**Comment by martinfowler** [comment]
Tried sonos-web again today, and now it works. No change in node, sonos-web, or ubuntu versions. Perhaps some package upgrades (but nothing that I would imagine would make a difference). No idea why it's started working again.

**Comment by puzio0** [comment]
> Tried sonos-web again today, and now it works. No change in node, sonos-web, or ubuntu versions. Perhaps some package upgrades (but nothing that I would imagine would make a difference). No idea why it's started working again.

I will try a full reinstall today. So far still receiving the same issue.

**Comment by mbriney** [comment]
I have the same problem as well, I've tried uninstall and re-installing to no avail.  Anyone find a work-around?

**Comment by the8127project** [comment]
If you go to:
/root/.sonos-web/src/sonos 
edit 
vi SonosNetwork.js
and then on line 715 add:
 if(queue.items)
            {
              queue.items.forEach((track, index) => { queue.items[index].queuePosition = index + 1; });
              return queue.items;
            }

then don't forget to do sonos-web stop and sonos-web start for this to become effective. 

I believe it has to do with playing radio and not music from the library - but is a guess.

**Comment by mbriney** [comment]
Hmm... that fix doesn't seem to solve it for me.  Still getting the same error.  

```js
  async _getQueue(deviceId) {
    const zone = this.devices.find(device => device.id === deviceId);
    if (!zone) { return null; }
    const queue = await zone.getQueue();
    if (queue) {
      if (queue.items) {
        queue.items.forEach((track, index) => { queue.items[index].queuePosition = index + 1; });
        return queue.items;
      }      
      //queue.items.forEach((track, index) => { queue.items[index].queuePosition = index + 1; });
      //return queue.items;
    }
    return null;
  }
```

**Comment by axi92** [comment]
I get the same error:

```
An unexpected error occurred while searching for your Sonos System: {"stack":"TypeError: Cannot read property 'forEach' of undefined
  at SonosNetwork._getQueue (/home/username/.sonos-web/src/sonos/SonosNetwork.js:715:19)
  at runMicrotasks (<anonymous>)
  at processTicksAndRejections (internal/process/task_queues.js:97:5)
  at async SonosNetwork._getAVTransportInfo (/home/username/.sonos-web/src/sonos/SonosNetwork.js:810:19)
  at async /home/username/.sonos-web/src/sonos/SonosNetwork.js:624:33
  at async Promise.all (index 2)
  at async /home/username/.sonos-web/src/sonos/SonosNetwork.js:623:9","message":"Cannot read property 'forEach' of undefined"} 
```

**Comment by j-davies** [comment]
This worked for me
https://github.com/Villarrealized/sonos-web/pull/65/files

**Comment by mattguide** [comment]
> 
> 
> This worked for me
> https://github.com/Villarrealized/sonos-web/pull/65/files

I made that change to my local SonosNetwork.js file and restarted `sonos-web`. It does get me past the 'forEach' error and loads the main web interface (yay!).  However, after the page loads the server immediately shuts down due to an error thrown by Service.js:
![image](https://user-images.githubusercontent.com/54867504/120552669-6159d300-c3ac-11eb-877e-5a9b687a83b6.png)

I am going to have to dig into it a bit more when I get a chance.

P.S. - I also tagged a bunch of Open issues above, since they all reference this error.
